### ISSUE TITLE
issues when building the develop branch with gcc8

### ISSUE THREAD

**Issue by Zeks** [yourWaifu/sleepy-discord#130]
`const Flag flags = Flag::VALID_EMBED;`

because of the `const` compiler cannot generate default assignment operator and fails

**Comment by Zeks** [comment]
Well... your current dev branch has at least 3 problems:
1) In embed.h at the very bottom `const Flag flags = Flag::VALID_EMBED;` doesn't allow newer gcc to generate assignment operator. 
2) nonstd::string_view doesn't have to_string which makes snowflake.h to fail compileing
3) even if above cases are sidestepped  this code raises assertion:
```

 SleepyDiscord::Embed embed;
 embed.title = "this is an embed";
 embed.description = "fics go here";
 sendMessage(message.channelID, "acceptable", embed);
```



`discord: /home/zeks/flipper/third_party/sleepy-discord/include/sleepy_discord/rapidjson/document.h:1735: int64_t rapidjson::GenericValue<Encoding, Allocator>::GetInt64() const [with Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>; int64_t = long int]: Assertion `data_.f.flags & kInt64Flag' failed.
`

**Comment by Zeks** [comment]
for the reference: master branch was compiling without problems.

**Comment by yourWaifu** [comment]
are you using cmake to configure your build or did you manually configure your build?

**Comment by yourWaifu** [comment]
Oh, I think I know what's the problem. Change this line

https://github.com/yourWaifu/sleepy-discord/blob/develop/include/sleepy_discord/message.h#L97

to

```cpp
std::string nonce = "";
```

**Comment by yourWaifu** [comment]
Hmm, that's interesting because it should just use a normal string according to the code. https://github.com/yourWaifu/sleepy-discord/blob/develop/include/sleepy_discord/nonstd/string_view.hpp#L91

**Comment by Zeks** [comment]
Yes, I know, but c++17 string_view doesn't have .to_string which is the problem for snowflake.h

**Comment by yourWaifu** [comment]
I haven't test this but I think this might fix it.

replace this

https://github.com/yourWaifu/sleepy-discord/blob/ac62cfed4c3d2d1101f3c625cfe4a41a19a07a76/include/sleepy_discord/snowflake.h#L19

with this
```
Snowflake(const nonstd::string_view & snow  ) : raw(snow.data(), snow.length()    ) {}
```

**Comment by Zeks** [comment]
I've done this locally:
`Snowflake(const nonstd::string_view & snow  ) : raw( snow) {}`
it compiles and seems to be working, but I have no idea if it's correct because I haven't fiddled with sv myself yet.
